code/drasil-code/Language/Drasil/Code/Imperative/Doxygen/Import.hs

```diff
@@ -22,10 +24,11 @@ verbosityToDoc :: Verbosity -> Doc
 verbosityToDoc Verbose = no
 verbosityToDoc Quiet = yes
 
-makeDoxConfig :: ProjName -> ProgData -> OptimizeChoice -> Verbosity -> Doc
+makeDoxConfig :: ProjName -> State GOOLState ProgData -> OptimizeChoice -> Verbosity -> Doc
 makeDoxConfig prog p opt v = 
   let fs = map filePath (filter (\f -> not (isEmpty $ modDoc $ fileMod f) && 
-             (isMainMod (fileMod f) || isHeader f)) (progMods p))
+             (isMainMod (fileMod f) || isHeader f)) (progMods (evalState p 
```

Comment:
It's usually not a good sign if you're using `evalState` in many places. It might be that really you do want `ProgData` here, and make it the responsibility of the caller to do the `evalState`.